### ISSUE TITLE
The LocalStreamEnvironment cannot be used when submitting a program through a client, or running in a TestEnvironment context. : help me

### ISSUE THREAD

**Issue by Ju-S** [naver/pinpoint#4618]
조언에 따라 버전을 호환되는 버전으로 바꿨는데 같은 오류가 발생하였습니다.
도와주시면 감사하겠습니다.
![error](https://user-images.githubusercontent.com/18659860/45533118-85a6a780-b831-11e8-81b3-78144387314e.PNG)

**Comment by minwoo-jung** [comment]
@Ju-S 
안녕하세요 

[문서의 설정 방법](https://github.com/naver/pinpoint/blob/master/doc/application-inspector.md#3-configuration)에 따라 설정을 해줘야 합니다.

특히 C항목의 설명대로 flink 인스턴스를 별도로 띄우고(cluster) job을 동작시킬경우(cluster 방식) flink.StreamExecutionEnvironment=server 설정을 추가해야합니다.

추가로 비슷한 [문의 이슈](https://github.com/naver/pinpoint/issues/4178)를 덧붙입니다.

**만약 가이드데로 설정했는데도 동일하게 문제가 발생해하면
flink 인스턴스 환경 구성 방법과 가이드 문서에 따라 변경한 설정 모두를 올려주세요. 같이 살펴봐드리겠습니다.^^**

**Comment by Ju-S** [comment]
먼저 192.168.0.52의 /home/user에 zookeeper-3.4.12와 scala-2.11.12, hadoop-2.8.3, flink-1.3.3을 설치하였습니다.
slave로 192.168.0.47, 192.168.0.53을 추가하고 각 slave에 동일한 flink를 설치하여 task manager에 뜨는 것을 확인하였습니다.
![flink task](https://user-images.githubusercontent.com/18659860/45725191-991c8e80-bbf4-11e8-9aa4-9a76c4dc285f.PNG)
pinpoint는 192.168.0.51에 설치하였으며, pinpoint/flink/src/main/resources/pinpoint-flink.properties와 pinpiont/flink/src/main/resources/hbase.properties를 가이드에 따라 수정하였습니다.
![pinpoint-flink](https://user-images.githubusercontent.com/18659860/45725236-daad3980-bbf4-11e8-9b1a-8346465383ee.PNG)
![hbase](https://user-images.githubusercontent.com/18659860/45725239-dd0f9380-bbf4-11e8-8063-41b66e22f0df.PNG)
이 상태로 job을 수행하기 위해 flink master인 192.168.0.52에 job을 돌린 결과입니다.

**Comment by minwoo-jung** [comment]
@Ju-S 
자세하게 현재 상황을 설명해주어서 상황을 파악하기 쉬워진 것 같습니다. ^^
전체적인 설정은 크게 문제가 없는것같습니다.

지금 상황에서 가장 크게 문제로 의심되는것은... 설정을 변경했으나 반영이 되지 않은것같습니다.
특히 pinpoint-flink.properties 파일의 수정사항이 반영되지 않은것같습니다.

일단 jar파일로 패키징한 job파일을 압축 풀어서 직접 확인해보는게 좋겠습니다.
`flink.StreamExecutionEnviroment=server` 가 아니라 `flink.StreamExecutionEnviroment=local`로 설정이 되있을것으로 예상됩니다.

### 해결 방법
**다시 maven package 명령어를 수행하고 job 파일을 새로 생성하고 실행해보세요.
컴파일 파일이 재활용 되지 않도록 clean 명령어도 꼭 함께 컴파일하는게  좋습니다.**

**Comment by Ju-S** [comment]
job실행문제는 해결되었습니다. 감사합니다.
그런데 pinpoint내의 application inspector에서 정보확인이 불가한데 서버를 zookeeper서버, flink서버, flink-slave서버, pinpoint서버(collector, hbase) 이렇게 최소4대의 서버를 구성해야 하나요? 저는 현재 위에 적혀있듯이 flink-master서버에 zookeeper를 설치하였습니다. 설정을 다시 보니 pinpoint서버와 연결될 만한 설정을 안되있었습니다. 어떤 부분을 수정해야 할까요?

**Comment by minwoo-jung** [comment]
가이드 문서의 `동작 구조`를 보면 동작을 이해는데 큰 도움이 됩니다.
pinpoint collector 서버에 설정해야하는 부분도 되어있는지 확인을 해봐야 됩니다.

collector에서 로그 레벨을 낮추고 `flink` 키워드로 검색해서 로그를 살펴보면 flink 서버에 데이터 전송여부를 확인할 수 있습니다.

**Comment by Ju-S** [comment]
혹시 로그 레벨을 낮추는 가이드가 있을까요?
pinpoint/collector/src/main/resources/pinpoint-collector.properties와 pinpoint/quickstart/collector/src/main/resouces/pinpiont-collector.properties의 내용을 아래와 같이 편집하였고, quickstart/bin/start-collector.sh로 실행하였습니다.
quickstart/collector의 pinpoint-collector.properties에는 flink.cluster관련 설정이 안되어있어 임의로 추가 하였습니다.
![collector properties](https://user-images.githubusercontent.com/18659860/45790924-427d8600-bcc2-11e8-9911-d8fc0c9aa7a1.PNG)

**Comment by minwoo-jung** [comment]
collector는 log4j를 사용하고 있습니다. log4j를 살펴보고 로그 레벨을 조정하시면 됩니다.

**Comment by Ju-S** [comment]
로그레벨을 TRACE로 변경한 후 아래와 같은 로그를 확인할 수 있었습니다. 연결이 된게 맞는거 아닌가요?
![log](https://user-images.githubusercontent.com/18659860/45791995-f719a680-bcc6-11e8-88ec-17ed2312ad68.PNG)

**Comment by Ju-S** [comment]
application-inspector의 설명을 봤을때 flink cluster는 스트림 정보 처리, zookeeper는 각서버의 정보 전달, collector는 agent의 정보를 flink서버에 전달 후, flink task manager가 hbase(저의 경우 pinpoint서버, collector서버와 동일)로 정보를 put하여 web에서 표시하는 것으로 이해하고 있습니다. 제가 이해한것이 맞나요?

**Comment by minwoo-jung** [comment]
네 이해하신게 맞습니다.

**Comment by Ju-S** [comment]
서버 목록에 agent가 1개일때는 원래 정보표시를 하지 않나요?

**Comment by minwoo-jung** [comment]
네 agent의 stat 정보를 이용하기때문에 agent가 1개일 경우 agent 1개만 이용해서 정보를 표시합니다.

질문하신 의도를 예측해보면...
agent가 1개일 경우 항상 min max 그래프가 avg 그래프와 비슷한 형태로 보여야 될것같은데 그렇게 보이지 않을 수 있습니다. min max의 값은 X축의 시간 단위 사이에 포함된 수치 중에 min, max값을 보여주기 때문에 avg 그래프와 비슷한 형태가 아닐 수 있습니다.

**Comment by Ju-S** [comment]
현재 상황으로는 no data collected로 표시되네요. log로는 확인했는데 위의 로그를 캡쳐한 사진이 있는데 agent정보를 교환한 로그가 아닌가요?

**Comment by minwoo-jung** [comment]
flink job 동작에는 특이사항은 없는지 flink에서 제대로 데이터를 받았는지... 확인해보는게 좋을것같습니다.
